28.03.2020

myPar opened the pull request.

Commit:
added class bin_tree with respective methods: for binary search tree

Commit:
added level printing for binary trees

Commit:
wrote "get same alphabet" task. Tested on different inputs based on binary search tree; TODO: free tree constructor for structure independent testing.

ddyak commented on trees/level_print.py:
```diff
@@ -0,0 +1,33 @@
+        else:
+            print(element.value, end=' ')
+
+            if not(element.left is None):
```
```if element.left is not None```

ddyak commented on trees/level_print.py:
```diff
@@ -0,0 +1,33 @@
+    queue = [root, "\n"]
+
+    while True:
+        element = queue.pop(0)
```
Удаление из начала списка - O(N)

Commit:
Start working on arithmetic expression parser task (printing arithmetic expressions):
added standard classes and exception classes.

myPar committed:
wrote correct exception information handling.

myPar committed:
wrote string parser (character list -> list of AST nodes)

myPar committed:
added reversed polish notation build
(previous and current implementation of the project supports exception processing)

myPar committed:
wrote build and print AST. Works properly. Began writing of correct infix notation checker (additional checker of the input expression correctness).
